pydatastructs/linear_data_structures/arrays.py

```diff
@@ -126,7 +126,7 @@ def __new__(cls, dtype=NoneType, *args, **kwargs):
     @classmethod
     def methods(cls):
         return ['__new__', '__getitem__',
-        '__setitem__', 'fill', '__len__']
+        '__setitem__', 'fill', '__len__', 'search']
```

Comment:
Please make it a function (like place in `algorithms.py` in `linear_data_structures`) instead of a class method?

Reply:
Pardon me, shall I include this directly? Since all this does is search in 1D arrays. I wanted to include this for dynamic arrays too, but that removal of items just makes a void location and does not shift the items in those void locations

Reply:
Regardless, I will include a jump search as it was mentioned as an idea in the issue. Again, for 1D static arrays. Is that fine?

Reply:
That should be fine. I am planning to add `ImplicitArray` so it would be useful to have search functions for that as well. And anyways, conceptually, search is an algorithm that just reads the data from an array. It's not responsible for updating the state of `OneDimensionalArray` class, so having it as a method is not of much use.

Reply:
Understood. So should I just move this change to algorithms.py along with addition of the jump search?

Reply:
Yes. Though let's focus only on one search algorithm for now. I think you are adding `binary_search`. Please make it that clear while naming the function. Like explicitly name the functions as `binary_search`, `jump_search`, `ternary_search`, etc.

Reply:
Actually, it is a hybrid. The search depends on the sorted order parameter. Positive means ascending sorted, zero is unsorted and negative is descending sorted.

Reply:
For 0 the search is simply linear, else it is binary. There is only a change of the comparison sign for descending array

Reply:
I don't think that's a good idea. The logic for each type of array should be encapsulated in a single function. The user can decide on their own what they might want to use. Please make separate functions for `binary_search`, `linear_search`, etc.

Reply:
If you prefer that I should separate this hybrid function, then I shall do that

Reply:
Ok understood

Reply:
Uh sorry to ask this. Should I make a new PR after doing as told?

Reply:
I pushed the new commit.